File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const lstack = [];
    const rstack = [];
    
    if(s[0]===")") return false;

    for(const el of s){
        if(el==="("){
            lstack.push(el);
        }else{
            if(lstack[lstack.length-1] === "("){
             lstack.pop();   
            }else{
             rstack.push(el);        
            }
        }
    }
    
    
    if(lstack.length === 0 &&  rstack.length === 0){
        return true
    }else{
        return false
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    
    for i, el in enumerate(s):
        if i == 0 and el == ")": return False
        
        
        if el ==")":
            for item in range(len(stack)):
                temp = stack.pop()
                if temp == "(":
                    break;
                else :
                    stack.append(temp)
        else :
            stack.append(el)
    

    answer = False if len(stack) else True
    
    
    return answer
    
    
    